providers/discord/src/airflow/providers/discord/notifications/discord.py

```diff
@@ -46,31 +49,163 @@ class DiscordNotifier(BaseNotifier):
     def __init__(
         self,
         discord_conn_id: str = "discord_webhook_default",
-        text: str = "This is a default message",
-        username: str = "Airflow",
-        avatar_url: str = ICON_URL,
+        text: str | None = None,
+        username: str | None = "Airflow",
+        avatar_url: str | None = ICON_URL,
         tts: bool = False,
+        success: bool | None = False,
+        failure: bool | None = False,
+        success_color: int = 0x00FF00,
```

Comment:
```suggestion
        *,
        success_color: int = 0x00FF00,
```

probably worth making the arguments key word only